### PR TITLE
Add `mc bucket version` command

### PR DESCRIPTION
**Pull request by poornas**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,13 +7,16 @@ require (
...
+	github.com/go-ini/ini v1.57.0 // indirect
 	github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
 	github.com/klauspost/compress v1.10.3
 	github.com/mattn/go-ieproxy v0.0.1
 	github.com/mattn/go-isatty v0.0.8
 	github.com/minio/cli v1.22.0
 	github.com/minio/minio v0.0.0-20200714163805-778e9c864f67
-	github.com/minio/minio-go/v7 v7.0.1
+	github.com/minio/minio-go v6.0.14+incompatible // indirect
+	github.com/minio/minio-go/v6 v6.0.57
```
Comment (kannappanr) [inline go.mod:18]: minio-go v6 should not show up here

File: cmd/bucket-main.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+var (
+	bucketFlags = []cli.Flag{}
+)
+
+var bucketCmd = cli.Command{
+	Name:            "bucket",
+	Usage:           "manage bucket",
+	Action:          mainBucket,
+	HideHelpCommand: true,
```
Comment (harshavardhana) [inline cmd/bucket-main.go:29]:
```suggestion
	Before: setGlobalsFromContext,
	HideHelpCommand: true,
```

File: cmd/bucket-version.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+ EXAMPLES:
+   1. Display bucket versioning status for bucket "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket
+
+   2. Enable versioning on bucket "mybucket" for alias "myminio".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket enable
+ 
+   3. Suspend versioning on bucket "mybucket" for alias "myminio".
+      {{.Prompt}} {{.HelpName}} myminio/mybucket suspend
+
```
Comment (harshavardhana) [inline cmd/bucket-version.go:55]: spurious space

File: cmd/main.go [hunk 4]
```diff
@@ -344,6 +344,7 @@ var appCmds = []cli.Command{
 	watchCmd,
 	policyCmd,
 	tagCmd,
+	bucketCmd,
```
Comment (harshavardhana) [inline cmd/main.go:347]: Missing entry in auto-complete codepath

File: cmd/bucket-version.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+	return string(jsonMessageBytes)
+}
+
+func (v bucketVersionMessage) String() string {
+	switch v.Op {
+	case "get":
+		msg := ""
+		switch v.Versioning.Status {
+		case "":
+			msg = fmt.Sprintf("%s versioning is disabled", v.URL)
```
Comment (kannappanr) [inline cmd/bucket-version.go:88]: I think the correct word is `un-versioned`

**Comment by kannappanr** [comment]
Please add `info` at the end to display the status. `mc bucket version alias/bucket info`

File: cmd/bucket-version.go [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+ 
+ USAGE:
+   {{.HelpName}} TARGET
+ 
+ FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+ EXAMPLES:
+   1. Display bucket versioning status for bucket "mybucket".
+	  {{.Prompt}} {{.HelpName}} myminio/mybucket info
```
Comment (harshavardhana) [inline cmd/bucket-version.go:48]:
```suggestion
	{{.Prompt}} {{.HelpName}} myminio/mybucket info
```

File: docs/minio-client-complete-guide.md [hunk 7]
```diff
@@ -1492,3 +1492,36 @@ Type      : file
...
   Content-Type: application/octet-stream
 ```
+
+<a name="version"></a>
+### Command `version`
+`version` manages bucket versioning
+
+```
+USAGE:
+   mc bucket version TARGET [enable | suspend | info]
```
Comment (harshavardhana) [inline docs/minio-client-complete-guide.md:1502]:
```suggestion
  mc bucket version TARGET [enable | suspend | info]
```

File: docs/minio-client-complete-guide.md [hunk 8]
```diff
@@ -1492,3 +1492,36 @@ Type      : file
...
+
+*Example: Display the versioning configuration for bucket `mybucket`*
+
+```
+mc bucket version myminio/mybucket info
+myminio/mybucket versioning status is enabled
+
+```
+*Example: Suspend versioning for bucket `mybucket`*
+
```
Comment (harshavardhana) [inline docs/minio-client-complete-guide.md:1523]:
```suggestion
```

File: go.sum [hunk 9]
```diff
@@ -277,6 +279,9 @@ github.com/minio/md5-simd v1.1.0 h1:QPfiOqlZH+Cj9teu0t9b1nTBfPbyTl16Of5MeuShdK4=
 github.com/minio/md5-simd v1.1.0/go.mod h1:XpBqgZULrMYD3R+M28PcmP0CkI7PEMzB3U77ZrKZ0Gw=
 github.com/minio/minio v0.0.0-20200714163805-778e9c864f67 h1:xMg87xvLhrw2jeE6xGx/uuFr2iIksfszITGdHhWURnA=
 github.com/minio/minio v0.0.0-20200714163805-778e9c864f67/go.mod h1:0WThPBMICOt5eEGT3XyaUy0cxYmoGhmJEvvRbiVqbYU=
+github.com/minio/minio-go v6.0.14+incompatible h1:fnV+GD28LeqdN6vT2XdGKW8Qe/IfjJDswNVuni6km9o=
```
Comment (harshavardhana) [inline go.sum:282]: looks like you are referencing minio-go without the version string..

File: cmd/client-s3.go [hunk 10]
```diff
@@ -2280,3 +2280,37 @@ func (c *S3Client) SetLifecycle(ctx context.Context, config *lifecycle.Configura
...
 }
+
+// GetVersioning - gets bucket versioning info.
+func (c *S3Client) GetVersioning(ctx context.Context) (config minio.BucketVersioningConfiguration, err *probe.Error) {
+	bucket, _ := c.url2BucketAndObject()
+	if bucket == "" {
+		return config, probe.NewError(BucketNameEmpty{})
+	}
+	var e error
+
```
Comment (harshavardhana) [inline cmd/client-s3.go:2291]:
```suggestion
```

File: cmd/client-s3.go [hunk 11]
```diff
@@ -2280,3 +2280,37 @@ func (c *S3Client) SetLifecycle(ctx context.Context, config *lifecycle.Configura
...
 	return nil
 }
+
+// GetVersioning - gets bucket versioning info.
+func (c *S3Client) GetVersioning(ctx context.Context) (config minio.BucketVersioningConfiguration, err *probe.Error) {
+	bucket, _ := c.url2BucketAndObject()
+	if bucket == "" {
+		return config, probe.NewError(BucketNameEmpty{})
+	}
+	var e error
```
Comment (harshavardhana) [inline cmd/client-s3.go:2290]:
```suggestion

	var e error
```

File: cmd/bucket-version.go [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+		op = strings.ToLower(args[1])
+	case "suspend":
+		op = strings.ToLower(args[1])
+	case "info":
+		op = strings.ToLower(args[1])
+	default:
+		fatalIf(probe.NewError(fmt.Errorf("Unknown argument %s passed", args[1])), "Invalid argument")
+	}
+
+	if op == "info" {
```
Comment (harshavardhana) [inline cmd/bucket-version.go:129]: perhaps a switch case would help?

File: cmd/bucket-version.go [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+		vConfig, e := client.GetVersioning(globalContext)
+		fatalIf(e, "Cannot get version info")
+		vMsg := bucketVersionMessage{
+			Op:     op,
+			Status: "success",
+			URL:    aliasedURL,
+		}
+		vMsg.Versioning.Status = vConfig.Status
+		vMsg.Versioning.MFADelete = vConfig.MFADelete
+		printMsg(vMsg)
```
Comment (harshavardhana) [inline cmd/bucket-version.go:139]:
```suggestion
		printMsg(vMsg)
		return nil
```

**Comment by kannappanr** [comment]
Please resolve the conflicts

**Comment by poornas** [comment]
> Please resolve the conflicts

done

**Pull request merged**